pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/render.go

```diff
@@ -346,9 +351,13 @@ func setTraitProperties(t *unstructured.Unstructured, traitName, namespace strin
 	if t.GetName() == "" {
 		t.SetName(traitName)
 	}
-
-	t.SetOwnerReferences([]metav1.OwnerReference{*ref})
-	t.SetNamespace(namespace)
+	// if application controller set the ownerReference and namespace already don't override it
+	if len(t.GetOwnerReferences()) == 0 {
+		t.SetOwnerReferences([]metav1.OwnerReference{*ref})
+	}
+	if len(t.GetNamespace()) == 0 {
```

Comment:
add annotation here.  and we should update namespace first before ownerRef, because we need to compare it with the namespace of AC